src/dos/dos_programs.cpp

```diff
@@ -46,9 +46,9 @@
 extern uint32_t floppytype;
 
 extern char autoexec_data[autoexec_maxsize];
-void CONFIG_ProgramStart(Program **make);
-void MIXER_ProgramStart(Program **make);
-void SHELL_ProgramStart(Program **make);
+std::unique_ptr<Program> CONFIG_ProgramCreate();
+std::unique_ptr<Program> MIXER_ProgramCreate();
+std::unique_ptr<Program> SHELL_ProgramCreate();
```

Comment:
Are these `*_ProgramCreate()` helper functions needed versus the template instantiation `ProgramCreate<...>` because these specific programs' class types (CONFIG, MIXER, and SHELL) aren't available in this scope?

Just for understand, not asking for changes.

Reply:
Yes. And they're classes with no separate declaration/definition, so there isn't a nice easy header to import.

Reply:
thanks!